providers/keycloak/src/airflow/providers/keycloak/auth_manager/routes/login.py

```diff
@@ -70,3 +70,38 @@ def login_callback(request: Request):
     secure = bool(conf.get("api", "ssl_cert", fallback=""))
     response.set_cookie(COOKIE_NAME_JWT_TOKEN, token, secure=secure)
     return response
+
+
+@login_router.get("/refresh")
+def refresh(request: Request) -> RedirectResponse:
+    """Refresh the token."""
+    client = KeycloakAuthManager.get_keycloak_client()
+    redirect_uri = request.url_for("refresh_callback")
+    auth_url = client.auth_url(redirect_uri=str(redirect_uri), scope="openid")
+    return RedirectResponse(auth_url)
```

Comment:
Sorry if I misunderstood something, I thought this would be in the public API, calling a `get_refresh_token_url` from the auth manager that would return what is currently `refresh_callback`, and we would simply operate a redirect from the public API to this refersH-callback url.

Then in the `refresh_callback` you're free to use the `GetUserDep` if you need. Bear in mind that a user with a valid refresh token and expired access token should be able to successfully access the endpoint I suppose.

Reply:
Thing is that core / users can't know the endpoint that is registered in the AuthManager, so I guess exposing a proxy route is more stable, leaving the flexibility to the auth manager to implement it under any url.

On the opposite if we skip the proxy route, it means that the auth manager needs to implement a url with a predefined pattern (otherwise UI and users of the API can't know which route it is).

I'm fine with both, since we do the proxy route for login / logout, I would recommend the same for refresh just for consistency. (unless we have a good reason not to go this way)

Reply:
I think following the same approach would be good. I will also include the core part as in the backend implementation. This will indeed enable using the endpoint directly from auth/refresh and from the public api similar to login/logout, we can also handle the next easily

In this case, UI should always use a proxy route to properly handle the unified implementation in the public api similar to others. That would be the only constraint for the UI. User can still call but that is up to user to manupilate the calls even though it won't be the practical approach 

I will also document that we should follow the same approach for all auth managers. Let me update the code accordingly

Thanks both! Cool discussion!